c10/core/TensorOptions.h

```diff
@@ -643,6 +643,8 @@ inline DeviceType computeDeviceType(DispatchKey tid) {
     return DeviceType::MSNPU;
   } else if (tid == DispatchKey::XLA) {
     return DeviceType::XLA;
+  } else if (tid == DispatchKey::XLAPreAutograd) {
+    return DeviceType::XLA;
```

Comment:
You shouldn't really ever hit this.

Reply:
It's triggered by a test calling legacy new constructor ;) . Here's backtrace without this change: 
```
#0  __cxxabiv1::__cxa_throw (obj=0x55555850f190, tinfo=0x7fffcebf9008 <typeinfo for c10::Error>,
    dest=0x7fffcd2c03bc <c10::Error::~Error()>)
    at /home/nwani/m3/conda-bld/compilers_linux-64_1560109574129/work/.build/x86_64-conda_cos6-linux-gnu/src/gcc/libstdc++-v3/libsupc++/eh_throw.cc:80
#1  0x00007fffcd626eb1 in c10::computeDeviceType (tid=c10::DispatchKey::XLAPreAutograd)
    at ../c10/core/TensorOptions.h:659
#2  0x00007fffcdbc4dcb in torch::utils::(anonymous namespace)::options (
    dispatch_key=c10::DispatchKey::XLAPreAutograd, scalar_type=c10::ScalarType::Float, device=...)
    at ../torch/csrc/utils/tensor_new.cpp:69
#3  0x00007fffcdbc51cc in torch::utils::(anonymous namespace)::new_with_sizes (
    dispatch_key=c10::DispatchKey::XLAPreAutograd, scalar_type=c10::ScalarType::Float, device=..., sizes=...)
    at ../torch/csrc/utils/tensor_new.cpp:110
#4  0x00007fffcdbcaafc in torch::utils::legacy_tensor_new (dispatch_key=c10::DispatchKey::XLAPreAutograd,
    scalar_type=c10::ScalarType::Float, args=(0, 3, 3), kwargs=0x0) at ../torch/csrc/utils/tensor_new.cpp:577
#5  0x00007fffcd2dc84f in torch::autograd::THPVariable_new (self=<Tensor at remote 0x7ffeb01c8550>,
    args=(0, 3, 3), kwargs=0x0) at ../torch/csrc/autograd/generated/python_variable_methods.cpp:689
```
@ezyang Looks like this change is necessary?